File: core/ParamsAndDtos/ProspectiveCandidateParams.cs
using System;
using core.Entities.Admin;
using core.Params;

namespace core.ParamsAndDtos
{
    public class ProspectiveCandidateParams: ParamPages
    {
        public int? Id {get; set;}
        public string CategoryRef {get; set;}
        public DateTime DateAdded { get; set; }
        public string CandidateNameLike {get; set;}
        public string Status {get; set;}
    }
}

File: client/src/app/shared/params/prospectiveCandidateParams.ts
export class prospectiveCandidateParams {
     id?=0;
     categoryRef?='';
     dateAdded?: Date;
     status= '';
     candidateNameLike?='';
     
     sort = "name";
     pageNumber = 1;
     pageSize = 10;
     search: string;
}